### PR TITLE
feat(proguard): Add release associations to proguard mapping files

### PR DESCRIPTION
**Pull request by buenaflor**
Ref: https://github.com/getsentry/team-mobile/issues/27

This implements the frontend side of the release association for proguard mapping files. The design is almost identical to sourcemaps.

Not sure if this is alright or needs refactoring but I took the Associations component and adjusted it a bit.

### CODE REVIEW

File: static/app/views/settings/projectProguard/associations.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      btnText={t(
+        '+ %s more',
+        associations.releases.length - associations.releases.slice(0, 4).length
+      )}
+      buttonProps={{
+        priority: 'default',
+        borderless: true,
+      }}
+    >
+      <NumericList>
```
Comment (priscilawebdev) [inline static/app/views/settings/projectProguard/associations.tsx:33]: we usually use the component https://github.com/getsentry/sentry/tree/master/static/app/components/list 😉

File: static/app/views/settings/projectProguard/associations.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        {associations.releases.map(release => (
+          <li key={release}>
```
Comment (priscilawebdev) [inline static/app/views/settings/projectProguard/associations.tsx:35]: should we use the component `ListItem` here? https://github.com/getsentry/sentry/tree/master/static/app/components/list/listItem

File: static/app/views/settings/projectProguard/associations.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  if (!associations.releases.length) {
+    return (
+      <NoAssociations>
+        {t('No releases associated with this proguard mapping file')}
+      </NoAssociations>
+    );
+  }
+
+  return (
+    <div>
```
Comment (priscilawebdev) [inline static/app/views/settings/projectProguard/associations.tsx:75]: do we need this `div`? can we use perhaps a `fragment`?

File: static/app/views/settings/projectProguard/index.tsx [hunk 4]
```diff
@@ -22,6 +22,15 @@ export type ProjectProguardProps = RouteComponentProps<{projectId: string}, {}>
   project: Project;
 };
 
+export type ProguardMappingAssociation = {
+  releases: string[];
+};
+
+export type AssociatedMapping = {
+  mapping: DebugFile | undefined;
+  releaseAssociation: DebugIdBundleAssociation[] | undefined;
```
Comment (priscilawebdev) [inline static/app/views/settings/projectProguard/index.tsx:31]:
```suggestion
  mapping?: DebugFile;
  releaseAssociation?: DebugIdBundleAssociation[];
```

File: static/app/views/settings/projectProguard/projectProguardRow.tsx [hunk 5]
```diff
@@ -14,16 +14,20 @@ import {IconClock, IconDelete, IconDownload} from 'sentry/icons';
...
   downloadRole: string;
   downloadUrl: string;
   mapping: DebugFile;
   onDelete: (id: string) => void;
   orgSlug: string;
+  associations?: ProguardMappingAssociation;
 };
 
 function ProjectProguardRow({
+  associations = {releases: []} as ProguardMappingAssociation,
```
Comment (priscilawebdev) [inline static/app/views/settings/projectProguard/projectProguardRow.tsx:30]: do we need to cast here?

File: static/app/views/settings/projectProguard/index.tsx [hunk 6]
```diff
@@ -42,6 +51,30 @@ function ProjectProguard({organization, location, router, params}: ProjectProgua
     }
   );
 
+  const associationsResults = useQueries({
```
Comment (priscilawebdev) [inline static/app/views/settings/projectProguard/index.tsx:54]: nice! I never used `useQueries` before

**Comment by priscilawebdev** [comment]
how does it look on dark mode and mobile devices? it would be nice to have a few screenshots or a recording video

**Comment by buenaflor** [comment]
@priscilawebdev here are some screenshots / video

Screenshot 2023-07-25 at 14 21 55
Screenshot 2023-07-25 at 14 16 22

File: static/app/views/settings/projectProguard/associations.tsx (hunk 3 above)
Comment (buenaflor) [reply static/app/views/settings/projectProguard/associations.tsx:75]: It seems like using `fragment` leads to the text being in a new line 
image

**Review by priscilawebdev — APPROVED** [review]
code looks good to me

**Pull request merged**